### PR TITLE
Apply the PR that cleans up the core module

### LINKED ISSUE #32: Apply the PR that cleans up the core module
Please pick https://github.com/CodeChain-io/codechain/pull/1933

### PR DESCRIPTION
**Pull request by Byeongjee**
It closes #32 
I couldn't cherry-pick  https://github.com/CodeChain-io/codechain/pull/1933/commits/25f1a6bb9db55a98804af43c6390b2f608c86389 failed, probably due to changes introduced by https://github.com/CodeChain-io/foundry/commit/c7db6519d9e9a06262d4cb88d7eae135972f08d6. 
I'm not sure whether we should introduce https://github.com/CodeChain-io/codechain/pull/1933/commits/25f1a6bb9db55a98804af43c6390b2f608c86389 to our current foundry code or not.

### CODE REVIEW

**Comment by sgkim126** [comment]
While there is a conflict if you cherry-pick the CodeChain's patch, you can apply it to Foundry with some fix.
The purpose of https://github.com/CodeChain-io/codechain/commit/25f1a6bb9db55a98804af43c6390b2f608c86389 is extracting the duplicated code to a function, and the purpose of https://github.com/CodeChain-io/foundry/commit/c7db6519d9e9a06262d4cb88d7eae135972f08d6 is changing the definition of the transactions root. They conflict because they fixed an adjacent code, but they are logically not related.
Would you try to fix the conflict? I think it's not different if you think the purpose of the conflicting commits.

**Comment by Byeongjee** [comment]
Thank you, I see. I'll work on it.

**Comment by Byeongjee** [comment]
I fixed the conflict. Could you review the change again? @sgkim126

File: core/src/block.rs [hunk 1]
```diff
@@ -248,23 +254,9 @@ impl<'x> OpenBlock<'x> {
...
-            })?;
-        }
+        self.close_impl(term_common_params)?;
 
-        let state_root = self.block.state.commit().map_err(|e| {
-            warn!("Encountered error on state commit: {}", e);
-            e
-        })?;
-        if self.block.header.transactions_root() == &BLAKE_NULL_RLP {
+        if self.block.header.transactions_root().is_zero() || self.block.header.transactions_root() == &BLAKE_NULL_RLP {
```
Comment (sgkim126) [inline core/src/block.rs]: We don't need `self.block.header.transactions_root().is_zero()`.

Comment (Byeongjee) [reply core/src/block.rs]: Sorry, I fixed it.

**Pull request merged**